src/serializer/modules.js

```diff
@@ -37,13 +37,17 @@ class ModuleTracer extends Tracer {
     this.requireSequence = [];
     this.logModules = logModules;
     this.uninitializedModuleIdsRequiredInEvaluateForEffects = new Set();
+    this.nestedRequiredModulesAndValues = [];
   }
 
   modules: Modules;
   evaluateForEffectsNesting: number;
   requireStack: Array<number | string | void>;
   requireSequence: Array<number | string>;
   uninitializedModuleIdsRequiredInEvaluateForEffects: Set<number | string>;
+  // We can't say that a module has been initialized if it was initialized in a
+  // evaluate for effects context until we know the effects are applied.
+  nestedRequiredModulesAndValues: Array<{| id: number | string, value: Value |}> = [];
```

Comment:
Since you initialize the field here, do you still need to do so in the constructor?